core/src/main/java/org/elasticsearch/cluster/health/ClusterShardHealth.java

```diff
@@ -69,6 +75,27 @@ public ClusterShardHealth(final int shardId, final IndexShardRoutingTable shardR
             }
         } else {
             computeStatus = ClusterHealthStatus.RED;
+            assert primaryRouting != null;
+            /**
+             * Normally, an inactive primary shard in an index should cause the cluster health to be RED.  However,
+             * there are exceptions where a health status of RED is inappropriate, namely in these scenarios:
+             *   1. Index Creation.  When an index is first created, the primary shards are in the initializing state, so
+             *      there is a small window where the cluster health is RED due to the primaries not being activated yet.
+             *      However, this leads to a false sense that the cluster is in an unhealthy state, when in reality, its
+             *      simply a case of needing to wait for the primaries to initialize.
+             *   2. When a cluster is in the recovery state, and the shard never had any allocation ids assigned to it,
+             *      which indicates the index was created and before allocation of the primary occurred for this shard,
+             *      a cluster restart happened.
+             *
+             * Here, we check for these scenarios and set the cluster health to YELLOW if any are applicable.
+             */
+            if (primaryRouting.unassignedInfo() != null) {
+                final UnassignedInfo.Reason reason = primaryRouting.unassignedInfo().getReason();
+                if (reason == UnassignedInfo.Reason.INDEX_CREATED
+                        || (reason == UnassignedInfo.Reason.CLUSTER_RECOVERED && activeAllocationIds.isEmpty())) {
+                    computeStatus = ClusterHealthStatus.YELLOW;
```

Comment:
green wouldn't make sense here because we know nothing of the replicas... and if the primaries are still unassigned, it means the replicas are too.  yellow would be the most logical step in moving away from red; the main reason for getting away from a red state anyway is to avoid triggers on cluster health causing administrators of ES clusters to get notifified and/or panic.

since the index is just being created at this point, with unavailable primaries and unavailable replicas, green wouldn't make sense.